github/repos.go

```diff
@@ -509,22 +509,21 @@ type Branch struct {
 
 // Protection represents a repository branch's protection.
 type Protection struct {
-	RequiredStatusChecks       *RequiredStatusChecks       `json:"required_status_checks"`
-	RequiredPullRequestReviews *RequiredPullRequestReviews `json:"required_pull_request_reviews"`
-	EnforceAdmins              *AdminEnforcement           `json:"enforce_admins"`
-	Restrictions               *BranchRestrictions         `json:"restrictions"`
+	RequiredStatusChecks *RequiredStatusChecks `json:"required_status_checks"`
+	EnforceAdmins        *AdminEnforcement     `json:"enforce_admins"`
+	Restrictions         *BranchRestrictions   `json:"restrictions"`
 }
 
 // ProtectionRequest represents a request to create/edit a branch's protection.
 type ProtectionRequest struct {
-	RequiredStatusChecks       *RequiredStatusChecks       `json:"required_status_checks"`
-	RequiredPullRequestReviews *RequiredPullRequestReviews `json:"required_pull_request_reviews"`
-	EnforceAdmins              bool                        `json:"enforce_admins"`
-	Restrictions               *BranchRestrictionsRequest  `json:"restrictions"`
+	RequiredStatusChecks *RequiredStatusChecks      `json:"required_status_checks"`
+	EnforceAdmins        bool                       `json:"enforce_admins"`
+	Restrictions         *BranchRestrictionsRequest `json:"restrictions"`
 }
 
 // RequiredStatusChecks represents the protection status of a individual branch.
 type RequiredStatusChecks struct {
+	// Deprecated: Use EnforceAdmins instead.
 	// Enforce required status checks for repository administrators. (Required.)
```

Comment:
You can't start the comment with "Deprecated:" first, because that breaks the Go style requirement of having documentation begin with identifier name. Put it second:

```
// Enforce required status checks for repository administrators. (Required.)
// Deprecated: Use EnforceAdmins instead.
```

Reference: https://github.com/golang/go/issues/10909#issuecomment-136492606.